src/triage/experiments/base.py

```diff
@@ -97,6 +107,8 @@ def __init__(
         profile=False,
         save_predictions=True,
     ):
+        experiment_kwargs = {k: v for k, v in locals().items() if k not in {'db_engine', 'config', 'self'}}
```

Comment:
Based on my experimentation it seems like it would be hard to make this do the same thing. What you posted appears to be creating a whitelist of arguments. What my example is taking all the args passed in and just blacklisting a couple so we don't have to continue updating this every time we add a kwarg.

Reply:
Sure. It's not so much that I was concerned about blacklisting/whitelisting, as much as ensuring that we supply correct signature parameters. You could think of it as a validation on what's a teensy bit hacky (using `locals`). You could certainly:

```
call_signature = inspect.signature(self.__class__).bind(**{
    key: value for (key, value) in locals().items() if key not in {'db_engine', 'config', 'self'}
})
```

(_i.e._ getting the variables in the same way).

:man_shrugging:

Reply:
So bind croaks in that case because it doesn't have 'config'. The 'bind' call has to be complete, it seems. To implement the blacklist it seems I can just bind with locals and then blacklist from the resulting BoundArguments object

```
        experiment_kwargs = {
            key: value
            for key, value in inspect.signature(self.__class__).bind(**(locals())).arguments['kwargs'].items()
            if key not in {'config', 'db_engine', 'self'}
        }
```

Reply:
Or just use `bind_partial`:

```
call_signature = inspect.signature(self.__class__).bind_partial(**{
    key: value for (key, value) in locals().items() if key not in {'db_engine', 'config', 'self'}
})
```

[[docs]](https://docs.python.org/3/library/inspect.html#inspect.Signature.bind_partial)

Reply:
Sure. This is so complicated now it should also go into the tracking module. The tracking module can't completely do this on its own because it doesn't have locals, but maybe just have the Experiment send locals() (with the blacklist applied) to the initialize_tracking_and_get_run_id, or even to a new function in triage.util.introspection!

Reply:
Yeah :crying_cat_face: 

Probably for the best tho!